tests/unit/preprocessor/_area/test_area.py

```diff
@@ -1445,5 +1447,21 @@ def test_time_dependent_volcello():
     assert cube.shape == cube.cell_measure('ocean_volume').shape
 
 
+@pytest.mark.parametrize('lazy', [True, False])
+def test_get_area_weights(lazy):
+    """Test _get_area_weights."""
+    cube = _create_sample_full_cube()
+    if lazy:
+        cube.data = cube.lazy_data()
+    weights = _get_area_weights(cube)
+    if lazy:
+        assert isinstance(weights, da.Array)
```

Comment:
Add a check that the chunks of the weights match the chunks of the cube?

Reply:
Done in https://github.com/ESMValGroup/ESMValCore/pull/2354/commits/e645a6244043d286957804e72ef134edcfdf1ac3